src/App/Application.h

```diff
@@ -69,6 +73,46 @@ enum class MessageOption {
     Throw, /**< Throw an exception. */
 };
 
+enum MeasureElementType {

```

Comment:
The main responsibility of Application is the handling with documents and it should be avoided to add further logic to it. So, all this code should be moved to a separate class e.g. MeasureManager. Since there aren't more instances needed of this class all methods could be static.

Or it can be a class member of Application which only has a method to access the manager. In this case it should be a std::unique_ptr to avoid to include its header in Application.h This is to considerably reduce build time in case the MeasureManager class will be extended later.